packages/studioCMS/src/integrations/studioCMSDashboard/routes/dashboard/components/CoreVitalPanelCard.astro

```diff
@@ -0,0 +1,278 @@
+        return Math.round(50 - ((lcp - 4) / (6 - 4)) * 50); // Fair to Good
+    }
+        return Math.round(0 - ((lcp - 6) / (10 - 6)) * 50); // Poor to Fair
+}
+
+function calculateInpAverage(inpValues: number[]): number {
+
+    const sum = inpValues.reduce((acc, curr) => acc + curr, 0);
+    const average = sum / inpValues.length;
+    
+    // Round average to two decimal places
+    return Math.round(average * 100) / 100;
+}
+
+function inpDataAverage(webVitalData: WebVitalsResponseItem[]): number {
+    const inpData: number[] = [];
+    if (webVitalData) {
+        for (const item of webVitalData) {
+            if (item.name === "INP") {
+                inpData.push(item.value);
+            }
+        }
+    }
+    return Math.floor(calculateInpAverage(inpData));
+}
+
+function calculateInpScore(inp: number): number {
+    if (inp <= 50) {
+        return 100; // Excellent
+    }if (inp <= 100) {
+        return Math.round(100 - ((inp - 50) / (100 - 50)) * 50); // Good to Excellent
+    }if (inp <= 200) {
+        return Math.round(50 - ((inp - 100) / (200 - 100)) * 50); // Fair to Good
+    }
+    return Math.round(0 - ((inp - 200) / (300 - 200)) * 50); // Poor to Fair
+}
+
+const inpColor = (inpData: number) => {
+    if (inpData <= 100) {
+        return "green";
+    } 
+    if (inpData > 100 && inpData <= 200) {
+        return "yellow";
+    } 
+    return "red";
+};
+
+const inpTrackColor = (inpData: number) => {
+    if (inpData <= 100) {
+        return "yellow";
+    } 
+    if (inpData > 100) {
+        return "red";
+    } 
+    return "red";
+};
+
+const lcpColor = (lcpData: number) => {
+    if (msToSeconds(lcpData) <= 2.5) {
+        return "green";
+    } 
+    if (msToSeconds(lcpData) > 2.5 && lcpData <= 4) {
+        return "yellow";
+    } 
+    return "red";
+};
+
+const lcpTrackColor = (lcpData: number) => {
+    if (msToSeconds(lcpData) <= 2.5) {
+        return "yellow";
+    } 
+    if (msToSeconds(lcpData) > 2.5) {
+        return "red";
+    } 
+    return "red";
+};
+
+const clsColor = (clsData: number) => {
+    if (clsData <= 0.25) {
+        return "green";
+    }
+    if (clsData > 0.25 && clsData <= 0.5) {
+        return "yellow";
+    }
+    return "red";
+}
+
+const clsTrackColor = (clsData: number) => {
+    if (clsData <= 0.25) {
+        return "yellow";
+    }
+    if (clsData > 0.25) {
+        return "red";
+    }
+    return "red";
+}
+
+---
+<div class="w-full grid max-sm:grid-cols-1 max-lg:grid-cols-2 lg:grid-cols-3 gap-4">
+    <Card>
+        <div slot="header" class="flex flex-row items-center gap-2">
+            <Icon slot="icon" name="speedometer" class:list={"lcp-icon"} />
+            <div class="text-xl font-bold">Largest Contentful Paint</div>
+        </div>
+        <div class="pt-2">
+            <ProgressBar 
+                value={calculateLcpScore(lcpDataAverage(lcpData))} 
+                label="LCP Data" 
+                class:list={"progressbar-lcp"}>
```

Comment:
Hmm does `class:list` support strings or is it just arrays? Might want to wrap this in an array!

Also goes for all the other `class:list`s

Reply:
I've never had an issue passing a simple string as long as its wrapped by `{ ... }`

Reply:
https://docs.astro.build/en/reference/directives-reference/#common-directives looks like it can take a bunch of different items and it just converts it to a simple string and adds to the class value

Reply:
[a599425](https://github.com/astrolicious/studiocms/pull/125/commits/a59942544adb965fdac4d691115d6f0af7108413) done